### PR TITLE
refactor: clean up orgs data layer

### PR DESCRIPTION
**Pull request by rudolfs**
Closes: https://github.com/radicle-dev/radicle-upstream/issues/2025.

### CODE REVIEW

**Review by geigerzaehler — COMMENTED** [review]
Great work!

File: ui/src/org/contract.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    throw new error.Error({
+      code: error.Code.OrgCreateNotFoundInInterfaceLogs,
+      message: "Org not found in interface logs",
+    });
+  }
+
+  return orgAddress;
+}
+
+export async function fetchGnosisSafeAddr(
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/src/org/contract.ts]: 9a115c3378f26725a486a6b2615bba4599747002

File: ui/src/org/contract.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  orgAddress: string,
+  provider: ethers.providers.Provider
+): Promise<string> {
+  const org = new ethers.Contract(orgAddress, orgAbi, provider);
+  const safeAddr: string = await org.owner();
+
+  return safeAddr.toLowerCase();
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/src/org/contract.ts]: For presentation reasons, I thought all lower case looked better in the UI. But perhaps we should show all addresses in their checksummed form? Dunno, I'm new to Ethereum. 🤷 

Removed it here: 9a115c3378f26725a486a6b2615bba4599747002.

File: ui/src/org/contract.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+export async function generateAnchorProjectTxData(
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/src/org/contract.ts:93]: 8655257c6971954106adfc05b3dd9630c278d658

File: ui/src/org/contract.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  const encodedProjectUrn = ethers.utils.zeroPad(
+    urn.parseIdentitySha1(projectUrn),
+    32
+  );
+  const encodedCommitHash = multihash.encode(
+    ethers.utils.arrayify(`0x${commitHash}`),
+    "sha1"
+  );
+
+  const orgContract = new ethers.Contract(checksummedGnosisSafeAddress, orgAbi);
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/src/org/contract.ts]: 796a94785eabd4a40ec50c60991e2a9974137035

File: ui/src/org/contract.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/src/org/contract.ts]: 2f40a50b6a1c6e04be3ff0ad41cb8839715749d8

**Review by geigerzaehler — APPROVED** [review]
:heart:

**Review by geigerzaehler — APPROVED** [review]

**Pull request merged**